### ISSUE TITLE
guard-de: Red / Green / Refactor / Commit

### ISSUE THREAD

**Issue by rking** [guard/guard#321]
# tl;dr

I'd like to add a custom command or two to guard. Is this possible?  Is it desirable?
# ww;tmm

("Wait wait, tell me more!")

I'm working on some new (to me) [REPL-based workflows](https://github.com/pry/pry/wiki/pry-de), in particular trying to make a sweet [red/green/refactor loop](https://github.com/pry/pry/wiki/pry-de#wiki-red-green-refactor). That loop is currently starting to shape up with `pry-rescue` automatically putting me into the REPL whenever I get a test failure. However, I'm writing to talk about the next step, _after_ each succesful {test fail ⇒ test pass} stage.

There are basically three courses of action:
1. Refactor what you did.
2. Start over again writing another failing test.
3. Commit to VCS.

For the refactor case, I imagine a guard command like `refactor` and would run:
    `${VISUAL:-${EDITOR:-vim}} $(git status --porcelain | cut -b4-) "$@"`

For new failing test, perhaps it would be a command that would open to the spec with the most recent timestamp? Maybe the syntax would be like:
    `another spec`
(which would descend through `spec/` looking for the most recent file. This would be generic enough that you could use it for editing files in entirely different workflows.)

For the commit to VCS, it could maybe just be ^Z and do some git commands in the background. Not sure if any guard integration would be useful (though if it showed a `git status` after every all-tests-passed it might be useful).

So I need your guidance on how to do this the right way.

Is there any kind of user-defined command mechanism in place?  If so, would anyone be interested in co-developing this small piece as a gem? (I'm happy to do most of the work, but if I had other input I could do it with more clarity.)

Should I instead focus on the RSpec runner, and leave Guard alone? I could accomplish those same 1/2/3 paths mentioned above by merely causing RSpec to do more than say "0 failures".

Thanks!
—☈
